h2o-core/src/main/java/hex/Model.java

```diff
@@ -713,6 +713,10 @@ private void addToUsedIfColumn(Set<String> usedColumns, Set<String> allColumns,
       return values;
     }
 
+    public DistributionFamily[] supportedDistributions() {
```

Comment:
Why is a list of supported distributions defined on `Parameter` object?

It seems the reason is that the output depends on the Parameters itself - eg.: https://github.com/h2oai/h2o-3/pull/5952/files#diff-1b45f7fd4b0c10d1411dcd1399033042b4916dca63fc6a5f881372d44b3f250dR165

This makes the overall semantic difficult to understand and it seems it is purpose build for AutoML use case and might be difficult to translate to other use cases as well.

It is definitely not obvious I need to fill all the other parameters and then call this function to see if I can also set the desired distribution. This duplicates what `init(false)` is meant for.

Instead of this custom function, can you set parameters including distribution, call init(false) and check if the '_distribution' had an error? or if there was any other error?

Reply:
The validations in the `init` functions are likely faulty in some cases (there are inconsistencies you pointed out) and we should fix those.

> Could it possibly be due to some reuse in RuleFit or IsolationForest? (rhetorical, I will see once I try it)
Not for Isolation Forest, possibly RuleFit but I doubt it - likely an omission.

I think we could split the PR into 2 steps:
- first adapt AutoML to respect the validations in the models (init method) - this is what users can do anyway
- fixing all validations in ModelBuilders with your findings (that doesn't have to be you who fixes this)